lahja/endpoint.py

```diff
@@ -729,4 +792,6 @@ def remove() -> None:
                     break
         finally:
             self._queues[event_type].remove(casted_queue)
+            if not self._queues[event_type]:
```

Comment:
This is a bug that should be backported, though I think I have a better solution using a combination of a `WeakValueDictionary` and a `WeakSet` that makes this auto-cleanup.